go/vt/mysqlctl/mysqld.go

```diff
@@ -688,7 +687,12 @@ func deleteTopDir(dir string) (removalErr error) {
 // executeMysqlCommands executes some SQL commands,
 // using the mysql command line tool.
 func (mysqld *Mysqld) executeMysqlCommands(connParams *sqldb.ConnParams, sql string) error {
-	return mysqld.executeMysqlScript(connParams, strings.NewReader(sql))
+	params, err := dbconfigs.WithCredentials(connParams)
```

Comment:
Those two usages make sense. The place where I ran into this was when creating a Backup, it stops and then starts mysql after the backup is finished.  After starting, it calls mysqld.Wait, which uses the root user.  https://github.com/youtube/vitess/blob/master/go/vt/mysqlctl/mysqld.go#L288